_docs/_latest/develop/dev/gcp-api.md

```diff
@@ -0,0 +1,220 @@
+
+Create a new foo API client file called foo.py in
+[google/cloud/forseti/common/gcp_api]({% link _docs/_latest/develop/reference/google.cloud.forseti.common.gcp_api.html %})
+package.
+
+## Create FooClient class to provide the entry point to the GCP API method
+
+1. Create FooClient class in foo.py.
+1. Initialize with the allowed quota and FooRepositoryClient (see next section).
+1. Create a get_(repository_client_property)s e.g. `get_instances()` as below,
+if the repository client has property `instances`. In the example below,
+the repository has list mixin, which will invoke the actual GCP API call.
+
+[google/cloud/forseti/common/gcp_api/cloudsql.py]({% link _docs/_latest/develop/reference/_modules/google/cloud/forseti/common/gcp_api/cloudsql.html %})
+
+```python
+class CloudsqlClient(object):
+    """CloudSQL Client."""
+
+    def __init__(self, global_configs, **kwargs):
+        """Initialize.
+
+        Args:
+            global_configs (dict): Global configurations.
+            **kwargs (dict): The kwargs.
+        """
+        max_calls, quota_period = api_helpers.get_ratelimiter_config(
+            global_configs, 'sqladmin')
+
+        self.repository = CloudSqlRepositoryClient(
+            quota_max_calls=max_calls,
+            quota_period=quota_period,
+            use_rate_limiter=kwargs.get('use_rate_limiter', True))
+
+    def get_instances(self, project_id):
+        """Gets all CloudSQL instances for a project.
+
+        Args:
+            project_id (int): The project id for a GCP project.
+
+        Returns:
+            list: A list of database Instance resource dicts for a project_id.
+            https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/instances
+        """
+
+        paged_results = self.repository.instances.list(project_id)
+        flattened_results = api_helpers.flatten_list_results(
+            paged_results, 'items')
+        LOGGER.debug('Getting all the cloudsql instances of a project,'
+                     ' project_id = %s, flattened_results = %s',
+                      project_id, flattened_results)
+        return flattened_results
+```
+
+## Create FooRepositoryClient class, and define the property method
+
+1. Create a FooRepositoryClient class in foo.py.
+1. Inherit from the `BaseRepositoryClient`.
+1. Initialize it with quota parameters.
+1. Define the property method named after the bar resource we want to interact
+with, e.g. bar(). This property method will initialize a \_FooBarRepository
+class (see next section), which will install the base GCP API functionalities
+(building requests, authentication, and API methods).
+
+
+[google/cloud/forseti/common/gcp_api/cloudsql.py]({% link _docs/_latest/develop/reference/_modules/google/cloud/forseti/common/gcp_api/cloudsql.html %})
+
+In this example, bar is ```instances```.
+
+```python
+class CloudSqlRepositoryClient(_base_repository.BaseRepositoryClient):
+    """Cloud SQL Admin API Respository."""
+
+    def __init__(self,
```

Comment:
Can we point to the generated source in /reference/ instead?

Reply:
I think that it's really helpful to reference the code and the explanations here in one place, so that people can get a good sense of the concepts and flow, even if the code goes through small changes.  Of course, any huge changes need to be updated here in the doc.  Is this okay?